### PR TITLE
Feature/add sorting and pagination

### PR DESCRIPTION
**Pull request by MrMizus**

### CODE REVIEW

File: src/app/components/category-products/category-products.component.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.dnone {
+    display: none !important;
```
Comment (gradzio) [inline src/app/components/category-products/category-products.component.scss:2]: there is aleady utility class for it i think it is called d-none

File: src/app/components/category-products/category-products.component.html [hunk 2]
```diff
@@ -45,17 +39,34 @@ <h5 class="mb-3">Categories</h5>
...
+            <div class="mb-3 mb-lg-0">
+              <p class="mb-0">
+                <span class="text-dark">{{ products.length }} </span> Products
+                found
+              </p>
+            </div>
+            <div class="d-flex mt-2 mt-lg-0">
+              <div [formGroup]="sort">
+                <!-- select option -->
+                <select class="form-select" formControlName="sortValue">
```
Comment (gradzio) [inline src/app/components/category-products/category-products.component.html:53]: would be nie to have these options as hardcoded observable in a component

File: src/app/components/category-products/category-products.component.ts [hunk 3]
```diff
@@ -1,30 +1,138 @@
...
-    this.category$
-  ]).pipe(map(([products, category]) => this._mapToProductQueryModels(products).filter((product) => product.categoryId === category.id).sort((a, b) => +b.featureValue - +a.featureValue)))
+    this.category$,
+    this.sortValue$,
+  ]).pipe(
+    map(([products, category, sortValue]) =>
+      this._mapToProductQueryModels(products)
+        .filter((product) => product.categoryId === category.id)
+        .sort((a, b) => {
+          if (sortValue === 'price-') {
```
Comment (gradzio) [inline src/app/components/category-products/category-products.component.ts]: you can simplify this code still. Nice you use the price / ratingValue as id - this is the way to go. you are only missing to extract information about asc / desc to another variable, then use property directly without any if statement

**Review by gradzio — CHANGES_REQUESTED** [review]

File: src/app/components/category-products/category-products.component.scss (hunk 1 above)
Comment (MrMizus) [reply src/app/components/category-products/category-products.component.scss:2]: I can't use "-" in ngClass

Comment (gradzio) [reply src/app/components/category-products/category-products.component.scss:2]: of course you can

**Review by gradzio — APPROVED** [review]

**Pull request merged**